src/proto/h1/io.rs

```diff
@@ -188,7 +189,9 @@ where
         if self.read_buf_remaining_mut() < next {
             self.read_buf.reserve(next);
         }
-        let mut buf = ReadBuf::uninit(&mut self.read_buf.bytes_mut()[..]);
+        let dst = self.read_buf.bytes_mut();
+        let dst = unsafe { &mut *(dst as *mut _ as *mut [MaybeUninit<u8>]) };
```

Comment:
That line looks threatening..

Can you please add a comment saying what it is doing?

Reply:
I don't actually know what it's doing ;)

I borrowed this from https://github.com/tokio-rs/tokio/blob/d78655337a68bded305782a8a8b4ac7be42aa6a7/tokio/src/io/util/read_buf.rs#L53-L55, which was the only updated use of `ReadBuf::uninit` I could find.

I would appreciate feedback who understand these APIs better than I do.

Reply:
@seanmonstar ?

Reply:
But let's see who introduced it 🙂

Reply:
https://github.com/tokio-rs/tokio/discussions/3121

Reply:
Thanks, added a comment summarizing that discussion.

Reply:
Perfect 😄